Library/Homebrew/test/formula_spec.rb

```diff
@@ -727,7 +727,7 @@ def test
     stub_formula_loader(f1)
 
     python = PythonRequirement.new
-    x11 = X11Requirement.new("x11", [:recommended])
+    x11 = (OS.mac? ? X11Requirement : XorgRequirement).new("x11", [:recommended])
```

Comment:
Glad you asked. In the future, I want to rewrite the X* requirements to use the OS extension system Mike introduced upstream, in the same manner as my work on the Java and FUSE requirements.

Presumably, this would imply:

1. Moving the XQuartz logic to extend/os/mac
2. Moving the linuxbrew/xorg/xorg handling stuff to extend/os/linux
3.  (Only on Linuxbrew, not upstream) adding XorgRequirement to the compat file to point to X11Requirement

What do you think of that plan?